js/src/logger.ts

```diff
@@ -92,6 +92,34 @@ import {
 import { lintTemplate } from "./mustache-utils";
 import { prettifyXact } from "@braintrust/core";
 
+// Fields that should be passed to the masking function
+// Note: "tags" field is intentionally excluded, but can be added if needed
+const REDACTION_FIELDS = [
+  "input",
+  "output",
+  "expected",
+  "metadata",
+  "context",
+] as const;
+
+/**
+ * Apply masking function to data and handle errors gracefully.
+ * If the masking function raises an exception, returns an error message with stack trace.
+ */
+function applyMaskingToField(
+  maskingFunction: (value: unknown) => unknown,
+  data: unknown,
+  fieldName: string,
+): unknown {
+  try {
+    return maskingFunction(data);
+  } catch (error) {
+    // Get the stack trace
+    const stack = error instanceof Error ? error.stack : String(error);
+    return `ERROR: Failed to mask data: ${stack}`;
```

Comment:
#nit I wonder if the error may include PII.

stupid example but one could imagine they might pass the secret to other functions which would then be part of the stack trace in some way:

<img width="781" height="238" alt="Image" src="https://github.com/user-attachments/assets/31211c46-9597-4990-9505-8de02e77aea5" />

safer yet, but I'll defer to you if we just do this or just return the simple message:

<img width="1018" height="900" alt="Image" src="https://github.com/user-attachments/assets/ab508dcd-806a-4470-8630-219365064eeb" />